webapp_v2/src/components/singlestudy/HomeView/StudyTreeView/index.tsx

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React, { useEffect, useMemo, useState } from 'react';
+import { Box, styled } from '@mui/material';
+import { StudyMetadata, VariantTree } from '../../../../common/types';
+import { StudyTree, getTreeNodes } from './utils';
+import { scrollbarStyle } from '../../../../theme';
+import { CIRCLE_RADIUS, colors, DCX, DCY, DEPTH_OFFSET, MIN_WIDTH, RECT_DECORATION, RECT_TEXT_WIDTH, RECT_X_SPACING, RECT_Y_SPACING, RECT_Y_SPACING_2, STROKE_WIDTH, TEXT_SIZE, TEXT_SPACING, TILE_SIZE_X, TILE_SIZE_Y, TILE_SIZE_Y_2, ZOOM_OUT, CURVE_OFFSET } from './treeconfig';
+
+export const SVGCircle = styled('circle')({
+  cursor: 'pointer',
+});
+export const SVGRect = styled('rect')({
+  cursor: 'pointer',
+});
+
+export const SVGText = styled('text')({
+  cursor: 'pointer',
+});
+
+interface Props {
+  study: StudyMetadata | undefined;
+  parents: Array<StudyMetadata>;
+  childrenTree: VariantTree | undefined;
+  onClick: (studyId: string) => void;
+}
+
+export default function CustomizedTreeView(props: Props) {
+  const { study, parents, childrenTree, onClick } = props;
+  const [studyTree, setStudyTree] = useState<StudyTree>();
+  const [hoverId, setHoverId] = useState<string>('');
+
+  const RECT_WIDTH = useMemo(() => {
+    if (studyTree === undefined) return 0;
+    const { drawOptions } = studyTree;
+    const { depth } = drawOptions;
+    return Math.max(TILE_SIZE_X * (depth + DEPTH_OFFSET), MIN_WIDTH);
+  }, [studyTree]);
+
+  const treeWidth = RECT_WIDTH + RECT_TEXT_WIDTH + RECT_X_SPACING;
+
+  const treeHeight = useMemo(() => {
+    if (studyTree === undefined) return 0;
+    const { drawOptions } = studyTree;
+    const { nbAllChildrens } = drawOptions;
+    return TILE_SIZE_Y * (nbAllChildrens + 1) + TILE_SIZE_Y_2;
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why this ? i guess there is all dependencies already ? if not maybe there is a problem